### PR TITLE
Holonotronic

### PR DESCRIPTION
**Pull request by Hellblazer**
singular cross existential ruleform authorizations for Phantasm.  Seemed natural to me to create the inverse using the inverse of the relationship, so that's what Phantasm maintains.

### CODE REVIEW

File: animations/src/main/java/com/chiralbehaviors/CoRE/meta/models/LocationModelImpl.java [hunk 1]
```diff
@@ -76,6 +85,54 @@ public void authorize(Aspect<Location> aspect, Attribute... attributes) {
         }
     }
 
+    /* (non-Javadoc)
+     * @see com.chiralbehaviors.CoRE.meta.NetworkedModel#authorize(com.chiralbehaviors.CoRE.ExistentialRuleform, com.chiralbehaviors.CoRE.relationship.Relationship, com.chiralbehaviors.CoRE.agency.Agency)
+     */
+    @Override
+    public void authorize(Location ruleform, Relationship relationship,
+                          Agency authorized) {
+        AgencyLocation a = new AgencyLocation(kernel.getCoreAnimationSoftware());
```
Comment (meleemistress) [inline animations/src/main/java/com/chiralbehaviors/CoRE/meta/models/LocationModelImpl.java]: So we don't have a LocationAgency object?

File: animations/src/main/java/com/chiralbehaviors/CoRE/meta/models/ProductModelImpl.java [hunk 2]
```diff
@@ -75,6 +84,57 @@ public void authorize(Aspect<Product> aspect, Attribute... attributes) {
         }
     }
 
+    /* (non-Javadoc)
+     * @see com.chiralbehaviors.CoRE.meta.NetworkedModel#authorize(com.chiralbehaviors.CoRE.ExistentialRuleform, com.chiralbehaviors.CoRE.relationship.Relationship, com.chiralbehaviors.CoRE.agency.Agency)
+     */
+    @Override
+    public void authorize(Product ruleform, Relationship relationship,
+                          Agency authorized) {
+        AgencyProduct a = new AgencyProduct(kernel.getCoreAnimationSoftware());
```
Comment (meleemistress) [inline animations/src/main/java/com/chiralbehaviors/CoRE/meta/models/ProductModelImpl.java]: If we don't have ProductAgency, make this unsupported, and we'll do everything in the AgencyModel

File: animations/src/main/java/com/chiralbehaviors/CoRE/meta/models/LocationModelImpl.java (hunk 1 above)
Comment (Hellblazer) [inline animations/src/main/java/com/chiralbehaviors/CoRE/meta/models/LocationModelImpl.java]: It's symmetric.  That would be silly to have.

File: animations/src/main/java/com/chiralbehaviors/CoRE/meta/models/ProductModelImpl.java (hunk 2 above)
Comment (Hellblazer) [inline animations/src/main/java/com/chiralbehaviors/CoRE/meta/models/ProductModelImpl.java]: No.  This is exactly what should be done.

Comment (meleemistress) [inline animations/src/main/java/com/chiralbehaviors/CoRE/meta/models/ProductModelImpl.java]: Wait a minute. Ignoring my style concerns, there's a bug here. Given agency A, Product P, and Relationship R, this code produces 2 AgencyProducts:
A R P
A R.Inverse P

That makes no sense. It should be P R.Inverse A

Comment (Hellblazer) [inline animations/src/main/java/com/chiralbehaviors/CoRE/meta/models/ProductModelImpl.java]: Pray tell how on earth A R.Inverse P is any differently stored in a database table than P R.Inverset A.

**Pull request merged**